File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-13/automation/2026-07-13-control-evidence-review-population-validation-tool.ts
/** Synthetic AC-27 validation helper for p03-full-005813; no network or file access. */
const organization = "Cobalt Harbor Systems";
const workingPaperContext = { stage: "intake-and-closure", evidenceCard: "EV-184-06", reviewLane: "PRIV-06", verificationPass: 6, workingPaperSequence: 5813, spineIds: ["org.cobalt-harbor", "control.approval-dual", "rel.ac27.evidence-184", "date.2026-07-13", "term.no-kio-claim"], rationale: "The bounded intake and AC-27 closure record were checked against the same synthetic evidence chain.", observation: "Population coverage records 48 synthetic privileged roles, with 47 timely attestations before compensating review.", canonicalValues: ["AC-27 requires preparer and reviewer role separation for each exception disposition."] };
type EvidenceRecord = { control_id: string; case_id: string; risk_id: string; evidence_set: number; reviewed: number; timely_attestations: number; residual_risk: string };
const expected: EvidenceRecord = { control_id: "AC-27", case_id: "CASE-260713-184", risk_id: "RISK-031", evidence_set: 184, reviewed: 48, timely_attestations: 47, residual_risk: "Medium" };
export function validate(record: EvidenceRecord): string[] { return Object.keys(expected).filter((key) => record[key as keyof EvidenceRecord] !== expected[key as keyof EvidenceRecord]); }
void organization; void workingPaperContext;

File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-13/automation/2026-07-13-control-evidence-review-population-validation-tool.rs
// Synthetic AC-27 validation helper for p03-full-004797; no network or file access.
// stage=intake-and-closure; card=EV-184-22; lane=PRIV-46; pass=2; working_paper=4797; spine=org.cobalt-harbor, evidence.set-184, rel.ac27.evidence-184, date.2026-07-13, term.no-kio-claim
// The bounded intake and AC-27 closure record were checked against the same synthetic evidence chain. Population coverage records 48 synthetic privileged roles, with 47 timely attestations before compensating review. Evidence set 184 contains 24 synthetic items: 12 access-review attestations, 6 approval records, 4 checksum manifests, and 2 exception records.
const ORGANIZATION: &str = "Cobalt Harbor Systems";
pub fn valid(control_id: &str, case_id: &str, risk_id: &str, evidence_set: u32, reviewed: u32, timely: u32, residual: &str) -> bool {
    let _organization = ORGANIZATION;
    control_id == "AC-27" && case_id == "CASE-260713-184" && risk_id == "RISK-031" && evidence_set == 184 && reviewed == 48 && timely == 47 && residual == "Medium"
}
